x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java

```diff
@@ -749,4 +727,21 @@ private void assertSearchableSnapshotStats(String indexName, boolean cacheEnable
         }
     }
 
+    private ByteSizeValue getCacheSizeForShard(ShardId shardId) {
+        ClusterStateResponse clusterStateResponse = client().admin().cluster().prepareState().setRoutingTable(true).setNodes(true).get();
+        ClusterState clusterStateResponseState = clusterStateResponse.getState();
+        String nodeId = clusterStateResponseState.getRoutingTable().shardRoutingTable(shardId).primaryShard().currentNodeId();
+        DiscoveryNode discoveryNode = clusterStateResponseState.nodes().get(nodeId);
+
+        final Settings nodeSettings = internalCluster().getInstance(Environment.class, discoveryNode.getName()).settings();
+        return CacheService.SNAPSHOT_CACHE_SIZE_SETTING.get(nodeSettings);
+    }
+
+    private int getShardCount(String indexName) {
```

Comment:
There's already a `getNumShards()` method available for this - it rreturn the number of primaries, replicas and data copies (basically what you did here)